File: src/courses/data/catalog/02-dsa/advanced-dsa/implementations/javascript/mst.js
/**
 * JavaScript Implementation: Minimum Spanning Tree (Kruskal & Prim)
 * Invariant: O(E log V) with Disjoint Set Union
 */
export class MinimumSpanningTreeKruskalPrim {
  constructor() {
    this.items = [];
  }

  // TODO: Implement verified invariant
  insert(val) {
    this.items.push(val);
  }

  get size() {
    return this.items.length;
  }
}

File: src/courses/data/catalog/02-dsa/advanced-dsa/implementations/python/mst.py
"""
Python Implementation: Minimum Spanning Tree (Kruskal & Prim)
Invariant: O(E log V) with Disjoint Set Union
"""
from typing import Generic, TypeVar, List, Optional

T = TypeVar('T')

class MinimumSpanningTreeKruskalPrim(Generic[T]):
    def __init__(self) -> None:
        self._items: List[T] = []

    # TODO: Implement core operational method
    def insert(self, value: T) -> None:
        self._items.append(value)

    def __len__(self) -> int:
        return len(self._items)
